Habitica/src/aura/TaskChangeStatusWindow/TaskChangeStatusWindowController.js

```diff
@@ -0,0 +1,28 @@
+({    
+    openModel: function(component, event, helper) {
+        component.set("v.isOpen", true);
+    },
+    
+    closeModel: function(component, event, helper) { 
+        component.set("v.isOpen", false);
+    },
+    
+    changeStatusAction: function(component, event, helper) {
+        var task = component.get("v.taskItem");
+        var newState = component.find("status").get("v.value");
+        task.Status__c = newState;
+        var updatedTaskToDoStatus = component.getEvent("updatedTaskStatus");
+        updatedTaskToDoStatus.setParams({ "taskItem": task });
+        updatedTaskToDoStatus.fire();
+        component.set("v.isOpen", false);
+    },
+    
+    loadOptions: function (component, event, helper) {
+        var opts = [
```

Comment:
you should get the options dynamically from the Status picklist's values on the server side by having a method in your Apex controller that iterates through the picklist metadata values and creates this collection.